### ISSUE TITLE
Reporting wrong line number

### ISSUE THREAD

**Issue by aperum** [ansible/ansible-lint#3435]
##### Summary

Ansible-lint is reporting the wrong line number for format warnings sometimes.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint 6.16.0 using ansible 2.14.5
```

- ansible OS package: community/ansible 7.5.0-2, community/ansible-core 2.14.5-1
- ansible installation method: one of source, pip, OS package
- ansible-lint installation method: pip 

##### STEPS TO REPRODUCE

```console (paste below)
$ cat playbooks/test.yml
─────────────────────────────────────────────
│ File: playbooks/test.yml
─────────────────────────────────────────────
   1   │ ---
   2   │
   3   │ - name: Test playbook
   4   │   hosts: localhost
   5   │
   6   │   tasks:
   7   │     - name: Test linter
   8   │       ansible.builtin.template:
   9   │         src: '{{ item }}'
  10   │         dest: '/{{ item}}'
  11   │         mode: '0755'
  12   │       loop:
  13   │         - foobar
─────────────────────────────────────────────

$ ~/.local/bin/ansible-lint playbooks/test.yml
WARNING  Listing 1 violation(s) that are fatal
jinja[spacing]: Jinja2 spacing could be improved: /{{ item}} -> /{{ item }} (warning)
playbooks/test.yml:9 Jinja2 template rewrite recommendation: `/{{ item }}`.

Read documentation for instructions on how to ignore specific rule violations.

              Rule Violation Summary
 count tag            profile rule associated tags
     1 jinja[spacing] basic   formatting (warning)

Passed with min profile: 0 failure(s), 1 warning(s) on 1 files.
```

**Comment by ssbarnea** [comment]
That is caused by limitation in used libraries as we do not have enough information to guess the exact line number, errors are reported at starting parent node.

I will close this as wontfix with the note that we will be more than happy to review PR that improves the behavior.
